gtsfm/frontend/matcher/superglue_matcher.py

```diff
@@ -0,0 +1,78 @@
+
+import numpy as np
+import torch
+
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.matcher.matcher_base import MatcherBase
+from thirdparty.SuperGluePretrainedNetwork.models.superglue import SuperGlue
+
+
+class SuperGlueMatcher(MatcherBase):
+    """Implements the SuperGlue matcher -- a pretrained graph neural network using attention."""
+
+    def __init__(self, use_cuda: bool = False):
+        super().__init__()
+        self._use_cuda = use_cuda
+
+        config = {
+            "weights": "outdoor",
+            "sinkhorn_iterations": 20,
+        }
+        self.superglue = SuperGlue(config).eval()
+
+    def match(
+        self,
+        keypoints_i1: Keypoints,
+        keypoints_i2: Keypoints,
+        descriptors_i1: np.ndarray,
+        descriptors_i2: np.ndarray,
+        img_height: int,
```

Comment:
@ayushbaid these 2 arguments are new, but i believe necessary. So I guess we'll add them to the other matcher files as well?

Reply:
They could be `Optional[int]` elsewhere, but cannot be optional here

Reply:
You are right. They need to be present here. However, we should have a consistent API for all matchers. Hence we should make it compulsory for all matchers.

I already have this change sitting in this branch: https://github.com/borglab/gtsfm/compare/feature/superglue

Reply:
well, thats kinda sad, should we continue reviewing this PR?

Reply:
@ayushbaid my 2 cents: I dont think its a good idea to make it compulsory for all matchers if most matchers are not going to need it. People reading the code would think "why is this even here?" unless they read the supergluematcher. Its kind of a case of [YAGNI](https://martinfowler.com/bliki/Yagni.html). Exceptions to the base API are okay where they make sense. If at some point we have many matchers that need it, the base class could have it. 
But I'll leave the choice to you. :)

Reply:
I think we need the exact same interface though, for inter-operability of the modules.

Reply:
The YAGNI article is cool -- thanks for sharing

Reply:
@ayushbaid i'll merge your branch into this one